src/json.hpp.re2c

```diff
@@ -205,50 +368,40 @@ struct is_compatible_object_type_impl<true, RealType, CompatibleObjectType>
         typename CompatibleObjectType::mapped_type>::value;
 };
 
-template<class RealType, class CompatibleObjectType>
+template<class BasicJson, class CompatibleObjectType>
```

Comment:
Yes, the worst part is that `BasicJson == Json` ... I'll rename it to `Json`, while trying to find a better name for the latter